pkg/network/protocols/http/statkeeper.go

```diff
@@ -73,15 +81,24 @@ func (h *StatKeeper) Process(tx Transaction) {
 // GetAndResetAllStats returns all the stats and resets the internal state.
 func (h *StatKeeper) GetAndResetAllStats() map[Key]*RequestStats {
 	h.mux.Lock()
-	defer h.mux.Unlock()
-
 	for _, tx := range h.incomplete.Flush(time.Now()) {
 		h.add(tx)
 	}
 
-	ret := h.stats // No deep copy needed since `h.stats` gets reset
+	// Rotate stats
+	stats := h.stats
 	h.stats = make(map[Key]*RequestStats)
-	return ret
+
+	// Rotate ConnectionAggregator
+	var aggregator *utils.ConnectionAggregator
+	if h.connectionAggregator == nil {
+		aggregator = h.connectionAggregator
+		h.connectionAggregator = utils.NewConnectionAggregator()
+	}
+	h.mux.Unlock()
```

Comment:
might be worth moving this logic into a separated helper function, and keep the mutex as defered

Reply:
Used an anonymous function to achieve. It felt odd to me to move the entire body (except one line) to a separate function